renderdoc/driver/gl/gl_driver.cpp

```diff
@@ -2484,6 +2484,60 @@ void WrappedOpenGL::SwapBuffers(void *windowHandle)
   }
 }
 
+void WrappedOpenGL::CreateVRAPITextureSwapChain(GLuint tex, GLenum textureType,
+                                                GLenum internalformat, GLsizei width, GLsizei height)
+{
+  GLResource res = TextureRes(GetCtx(), tex);
+  ResourceId id = GetResourceManager()->RegisterResource(res);
+
+  if(m_State >= WRITING)
+  {
+    Chunk *chunk = NULL;
+
+    {
+      SCOPED_SERIALISE_CONTEXT(GEN_TEXTURE);
+      Serialise_glGenTextures(1, &tex);
+
+      chunk = scope.Get();
+    }
+
+    GLResourceRecord *record = GetResourceManager()->AddResourceRecord(id);
+    RDCASSERT(record);
+
+    record->AddChunk(chunk);
+  }
+  else
+  {
+    GetResourceManager()->AddLiveResource(id, res);
+  }
+
+  GLenum format = eGL_RGBA;
+  GLenum type = eGL_UNSIGNED_BYTE;
+
+  m_Textures[id].resource = res;
+  m_Textures[id].curType = eGL_TEXTURE_2D;
+  m_Textures[id].dimension = 2;
+  m_Textures[id].width = width;
+  m_Textures[id].height = height;
+  m_Textures[id].depth = 1;
+  m_Textures[id].samples = 1;
+  m_Textures[id].creationFlags = TextureCategory::NoFlags;
+  m_Textures[id].internalFormat = internalformat;
```

Comment:
I think the existing code should handle filling out the `m_Textures` below in `Common_glTextureImage3DEXT` and `Common_glTextureImage2DEXT`, right? So we can simplify the code by removing this and letting it be handled in the same path, to reduce the number of places that need to change if it's updated in future.